### PR TITLE
[11.0] [ADD] Search Partner by Phone & Mobile

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Search Partner Phone/Mobile/Email
===========================

* This module searches a partner email and phone & mobile number.

@JayVora-SerpentCS Could you please review it?

### CODE REVIEW

File: partner_phone_search/README.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: https://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===========================
+Search Partner Phone/Mobile
```
Comment (JayVora-SerpentCS) [inline partner_phone_search/README.rst]: Search Partner by Phone/Mobile/Email
Add search by email too

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]
Add test cases too

**Comment by nikul-serpentcs** [comment]
@JayVora-SerpentCS Added unit test case and email, ready to review.

**Review by gurneyalex — APPROVED** [review]

File: partner_phone_search/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        if name:
+            domain = ['|', '|', ('phone', operator, name),
```
Comment (pedrobaeza) [inline partner_phone_search/models/res_partner.py]: I remember to have problems implementing something similar in 

https://github.com/OCA/l10n-spain/blob/dc1d2add9990a27f711901149d2a9fbfb9dda24c/l10n_es_partner/models/res_partner.py#L39

And I finally gave up giving precedence to my query.

**Review by hieulucky111 — APPROVED** [review]

**Comment by nikul-serpentcs** [comment]
@pedrobaeza Improved searching function as your comment, Could you please review it?

File: partner_phone_search/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        if not args:
+            args = []
+        if name:
+            domain = ['|', '|', ('phone', operator, name),
+                      ('mobile', operator, name), ('email', operator, name)
```
Comment (pedrobaeza) [inline partner_phone_search/models/res_partner.py:16]: Why including email here? It's not the scope nor the spec of the module

Comment (nikul-serpentcs) [reply partner_phone_search/models/res_partner.py:16]: Ok, Than I will adding an email discerption in scope & spec, Right?

**Review by JayVora-SerpentCS — APPROVED** [review]

File: partner_phone_search/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 - TODAY Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Search Partner Phone/Mobile/Email',
+    'version': '11.0.1.0.0',
+    'category': 'Extra Tools',
+    'summary': 'Partner Search by Phone/Mobile/Email',
+    'author': '''Serpent Consulting Services Pvt. Ltd.,
```
Comment (pedrobaeza) [inline partner_phone_search/__manifest__.py]: Don't use triple quote for this, or it will show a lot of empty spaces

**Comment by nikul-serpentcs** [comment]
@pedrobaeza Improved Code as per comment and It's Ready for Review. it's done then Could you please move Forward.

**Pull request merged**